plugins/modules/ndo_virtual_port_channel_interface.py

```diff
@@ -0,0 +1,383 @@
+            ["state", "present", ["template", "virtual_port_channel_interface"]],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    virtual_port_channel_interface = module.params.get("virtual_port_channel_interface")
+    virtual_port_channel_interface_uuid = module.params.get("virtual_port_channel_interface_uuid")
+    description = module.params.get("description")
+    node_1 = module.params.get("node_1")
+    node_2 = module.params.get("node_2")
+    interfaces_node_1 = module.params.get("interfaces_node_1")
+    if interfaces_node_1:
+        interfaces_node_1 = ",".join(interfaces_node_1)
+    interfaces_node_2 = module.params.get("interfaces_node_2")
+    if interfaces_node_2:
+        interfaces_node_2 = ",".join(interfaces_node_2)
+    interface_policy_group = module.params.get("interface_policy_group")
+    interface_policy_group_uuid = module.params.get("interface_policy_group_uuid")
+    interface_descriptions = module.params.get("interface_descriptions")
+    state = module.params.get("state")
+
+    ops = []
+    match = None
+
+    mso_template = MSOTemplate(mso, "fabric_resource", template)
+    mso_template.validate_template("fabricResource")
+    object_description = "Virtual Port Channel Interface"
+
+    path = "/fabricResourceTemplate/template/virtualPortChannels"
+    existing_virtual_port_channel_interfaces = mso_template.template.get("fabricResourceTemplate", {}).get("template", {}).get("virtualPortChannels", [])
+    if virtual_port_channel_interface or virtual_port_channel_interface_uuid:
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            existing_virtual_port_channel_interfaces,
+            [KVPair("uuid", virtual_port_channel_interface_uuid) if virtual_port_channel_interface_uuid else KVPair("name", virtual_port_channel_interface)],
+        )
+        if match:
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = existing_virtual_port_channel_interfaces
+
+    if state == "present":
+
+        if interface_policy_group and not interface_policy_group_uuid:
+            fabric_policy_template = MSOTemplate(mso, "fabric_policy", interface_policy_group.get("template"))
+            fabric_policy_template.validate_template("fabricPolicy")
+            interface_policy_group_uuid = fabric_policy_template.get_interface_policy_group_uuid(interface_policy_group.get("name"))
+
+        if match:
+            if virtual_port_channel_interface and match.details.get("name") != virtual_port_channel_interface:
+                ops.append(dict(op="replace", path="{0}/{1}/name".format(path, match.index), value=virtual_port_channel_interface))
+                match.details["name"] = virtual_port_channel_interface
+
+            if description and match.details.get("description") != description:
```

Comment:
how do we rest to None? should this be 
```suggestion
            if description is not None and match.details.get("description") != description:
```

Reply:
same would apply for any other string that could be reset to ""

Reply:
you're right, will change that